### DISCUSSION TITLE
John Snow Labs Spark-NLP 4.4.0: New BART for Text Translation & Summarization, new ConvNeXT Transformer for Image Classification, new Zero-Shot Text Classification by BERT, more than 4000+ state-of-the-art models, and many more!

### DISCUSSION THREAD

**Discussion by maziyarpanahi** [JohnSnowLabs/spark-nlp#13754] [Announcement]
---------------
:loudspeaker: Overview
---------------

We are thrilled to announce the release of Spark NLP 🚀 4.4.0! This release includes new features such as a New **BART** for NLG, translation, and comprehension; a new **ConvNeXT** Transformer for Image Classification, a new **Zero-Shot Text Classification** by **BERT**, **4000+** new **state-of-the-art models**, and more enhancements and bug fixes.

We want to thank our community for their valuable feedback, feature requests, and contributions. Our [Models Hub](https://nlp.johnsnowlabs.com/models?edition=Spark+NLP) now contains over 17,000+ free and truly open-source models & pipelines. 🎉

### **Spark NLP** has a new home! https://sparknlp.org is where you can find all the documentation, models, and demos for Spark NLP. It aims to provide valuable resources to anyone interested in 100% open-source NLP solutions by using **Spark NLP** 🚀.

----------------
:fire: New Features
----------------

### ConvNeXT Image Classification (By Facebook)

**NEW:** Introducing **ConvNextForImageClassification** annotator in Spark NLP 🚀. `ConvNextForImageClassification` can load `ConvNeXT` models that compete favorably with Transformers in terms of accuracy and scalability, achieving 87.8% ImageNet top-1 accuracy and outperforming Swin Transformers on COCO detection and ADE20K segmentation, while maintaining the simplicity and efficiency of standard ConvNets.

This annotator is compatible with all the models trained/fine-tuned by using `ConvNextForImageClassification` for **PyTorch** or `TFConvNextForImageClassification` for **TensorFlow** models in HuggingFace 🤗

![](https://i.imgur.com/jk2fVUT.png)

> [A ConvNet: ImageNet-1K classification results for • ConvNets and ◦ vision Transformers. Each bubble’s area is proportional to FLOPs of a variant in a model family.](https://arxiv.org/pdf/2201.03545.pdf) by Zhuang Liu, Hanzi Mao, Chao-Yuan Wu, Christoph Feichtenhofer, Trevor Darrell, Saining Xie.

### BART for NLG, Translation, and Comprehension (By Facebook)

**NEW:** Introducing **BartTransformer** annotator in Spark NLP 🚀. `BartTransformer` can load BART models fine-tuned for tasks like summarizations. 

This annotator is compatible with all the models trained/fine-tuned by using `BartForConditionalGeneration` for **PyTorch** or `TFBartForConditionalGeneration ` for **TensorFlow** models in HuggingFace 🤗

The abstract explains that Bart uses a standard seq2seq/machine translation architecture, similar to BERT's bidirectional encoder and GPT's left-to-right decoder. The pretraining task involves randomly shuffling the original sentences and replacing text spans with a single mask token. BART is effective for text generation and comprehension tasks, matching RoBERTa's performance with similar training resources on GLUE and SQuAD. It also achieves new state-of-the-art results on various summarization, dialogue, and question-answering tasks with gains of up to 6 ROUGE.

![](https://i.imgur.com/7xBdki1.png)
> The Bart model was proposed in [BART: Denoising Sequence-to-Sequence Pre-training for Natural Language Generation, Translation, and Comprehension](https://arxiv.org/abs/1910.13461) by Mike Lewis, Yinhan Liu, Naman Goyal, Marjan Ghazvininejad, Abdelrahman Mohamed, Omer Levy, Ves Stoyanov, and Luke Zettlemoyer

### Zero-Shot for Text Classification by BERT

**NEW:** Introducing **BertForZeroShotClassification** annotator for Zero-Shot Text Classification in Spark NLP 🚀. You can use the `BertForZeroShotClassification` annotator for text classification with your labels! :100: 

> Zero-Shot Learning (ZSL): Traditionally, ZSL most often referred to a fairly specific type of task: learning a classifier on one set of labels and then evaluating on a different set of labels that the classifier has never seen before. Recently, especially in NLP, it's been used much more broadly to get a model to do something it wasn't explicitly trained to do. A well-known example of this is in the [GPT-2 paper](https://pdfs.semanticscholar.org/9405/cc0d6169988371b2755e573cc28650d14dfe.pdf) where the authors evaluate a language model on downstream tasks like machine translation without fine-tuning on these tasks directly.

Let's see how easy it is to just use any set of labels our trained model has never seen via the `setCandidateLabels()` param:

```python
zero_shot_classifier = BertForZeroShotClassification \
    .pretrained() \
    .setInputCols(["document", "token"]) \
    .setOutputCol("class") \
    .setCandidateLabels(["urgent", "mobile", "travel", "movie", "music", "sport", "weather", "technology"])

```

For Zero-Short Multi-class Text Classification:
```text
+----------------------------------------------------------------------------------------------------------------+--------+
|result                                                                                                          |result  |
+----------------------------------------------------------------------------------------------------------------+--------+
|[I have a problem with my iPhone that needs to be resolved asap!!]                                              |[mobile]|
|[Last week I upgraded my iOS version and ever since then my phone has been overheating whenever I use your app.]|[mobile]|
|[I have a phone and I love it!]                                                                                 |[mobile]|
|[I want to visit Germany and I am planning to go there next year.]                                              |[travel]|
|[Let's watch some movies tonight! I am in the mood for a horror movie.]                                         |[movie] |
|[Have you watched the match yesterday? It was a great game!]                                                    |[sport] |
|[We need to hurry up and get to the airport. We are going to miss our flight!]                                  |[urgent]|
+----------------------------------------------------------------------------------------------------------------+--------+
```

For Zero-Short Multi-class Text Classification:
```text
+----------------------------------------------------------------------------------------------------------------+-----------------------------------+
|result                                                                                                          |result                             |
+----------------------------------------------------------------------------------------------------------------+-----------------------------------+
|[I have a problem with my iPhone that needs to be resolved asap!!]                                              |[urgent, mobile, movie, technology]|
|[Last week I upgraded my iOS version and ever since then my phone has been overheating whenever I use your app.]|[urgent, technology]               |
|[I have a phone and I love it!]                                                                                 |[mobile]                           |
|[I want to visit Germany and I am planning to go there next year.]                                              |[travel]                           |
|[Let's watch some movies tonight! I am in the mood for a horror movie.]                                         |[movie]                            |
|[Have you watched the match yesterday? It was a great game!]                                                    |[sport]                            |
|[We need to hurry up and get to the airport. We are going to miss our flight!]                                  |[urgent, travel]                   |
+----------------------------------------------------------------------------------------------------------------+-----------------------------------+
```

----------------
:star::bug: Improvements & Bug Fixes
----------------
* Add a new `nerHasNoSchema` param for NerConverter when labels coming from NerDLMOdel and NerCrfModel don't have any schema
* Set custom entity name in Data2Chunk via `setEntityName` param
* Fix loading `WordEmbeddingsModel` bug when loading a model from S3 via the `cache_folder` config
* Fix the `WordEmbeddingsModel` bug failing when it's used with `setEnableInMemoryStorage` set to `True` and LightPipeline
* Remove deprecated parameter `enablePatternRegex` from `EntityRulerApproach` & `EntityRulerModel`
* Welcoming 3 new **Databricks** runtimes to our Spark NLP family:
  * Databricks 12.2 LTS
  * Databricks 12.2 LTS ML
  * Databricks 12.2 LTS ML GPU  
* Deprecate Python 3.6 in Spark NLP 4.4.0

--------------------
:floppy_disk: Models
--------------------
Spark NLP 4.4.0 comes with more than **4300+** new state-of-the-art pre-trained transformer models in multi-languages.  

### Featured Models

| Model                | Name               | Lang |  
|:---------------------|:-------------------|:---|
| BertForZeroShotClassification | [bert_base_cased_zero_shot_classifier_xnli](https://sparknlp.org/2023/04/05/bert_base_cased_zero_shot_classifier_xnli_en.html) |      `en`
| ConvNextForImageClassification | [image_classifier_convnext_tiny_224_local](https://sparknlp.org/2023/03/28/image_classifier_convnext_tiny_224_local_en.html) |      `en`
| BartTransformer | [distilbart_xsum_12_6](https://sparknlp.org/2023/04/09/distilbart_xsum_12_6_en.html) |      `en`
| BartTransformer | [bart_large_cnn](https://sparknlp.org/2023/04/09/bart_large_cnn_en.html) |      `en`
| BertForQuestionAnswering | [bert_qa_case_base](https://sparknlp.org/2023/04/04/bert_qa_case_base_en.html) |      `en`
| HubertForCTC | [asr_swin_exp_w2v2t_nl_hubert_s319](https://sparknlp.org/2023/04/01/asr_swin_exp_w2v2t_nl_hubert_s319_nl.html) |      `nl`
| BertForTokenClassification | [bert_token_classifier_base_chinese_ner](https://sparknlp.org/2023/03/22/bert_token_classifier_base_chinese_ner_zh.html) |      `zh`

### :notebook: New Notebooks

| Notebooks | Colab |
| :-----------| :------|
|Zero-Shot Text Classification|[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/JohnSnowLabs/spark-nlp/blob/master/examples/python/annotation/text/english/zero-shot%20text%20classification/Zero_Shot_Text_Classification_by_BERT.ipynb)
|Document Summarization with BART|[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/JohnSnowLabs/spark-nlp/blob/master/examples/python/annotation/text/english/text-summarization/Text_Summarization_with_BART.ipynb)

- You can visit [Import Transformers in Spark NLP](https://github.com/JohnSnowLabs/spark-nlp/discussions/5669)
- You can visit [Spark NLP Examples](https://github.com/JohnSnowLabs/spark-nlp/tree/master/examples) for 100+ examples

----------------
:book: Documentation
----------------
* [Import models from TF Hub & HuggingFace](https://github.com/JohnSnowLabs/spark-nlp/discussions/5669) 
* [Spark NLP Notebooks](https://github.com/JohnSnowLabs/spark-nlp/tree/master/examples)
* [Models Hub](https://nlp.johnsnowlabs.com/models?edition=Spark+NLP) with new models
* [Spark NLP Articles](https://medium.com/spark-nlp)
* [Spark NLP in Action](https://nlp.johnsnowlabs.com/demo)
* [Spark NLP Documentation](https://nlp.johnsnowlabs.com/docs/en/quickstart)
* [Spark NLP Scala APIs](https://nlp.johnsnowlabs.com/api)
* [Spark NLP Python APIs](https://nlp.johnsnowlabs.com/api/python)

### Community support

- [Slack](https://join.slack.com/t/spark-nlp/shared_invite/zt-198dipu77-L3UWNe_AJ8xqDk0ivmih5Q) For live discussion with the Spark NLP community and the team
- [GitHub](https://github.com/JohnSnowLabs/spark-nlp) Bug reports, feature requests, and contributions
- [Discussions](https://github.com/JohnSnowLabs/spark-nlp/discussions) Engage with other community members, share ideas,
  and show off how you use Spark NLP!
- [Medium](https://medium.com/spark-nlp) Spark NLP articles
- [YouTube](https://www.youtube.com/channel/UCmFOjlpYEhxf_wJUDuz6xxQ/videos) Spark NLP video tutorials

---------------
Installation
---------------
**Python**

```shell
#PyPI

pip install spark-nlp==4.4.0
```

**Spark Packages**

**spark-nlp** on Apache Spark 3.0.x, 3.1.x, 3.2.x, and 3.3.x (Scala 2.12):

```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp_2.12:4.4.0

pyspark --packages com.johnsnowlabs.nlp:spark-nlp_2.12:4.4.0
```

**GPU**

```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-gpu_2.12:4.4.0

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-gpu_2.12:4.4.0
```

**Apple Silicon (M1 & M2)**
```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-silicon_2.12:4.4.0

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-silicon_2.12:4.4.0
```

**AArch64**
```shell
spark-shell --packages com.johnsnowlabs.nlp:spark-nlp-aarch64_2.12:4.4.0

pyspark --packages com.johnsnowlabs.nlp:spark-nlp-aarch64_2.12:4.4.0
```

**Maven**

**spark-nlp** on Apache Spark 3.0.x, 3.1.x, 3.2.x, and 3.3.x:

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp_2.12</artifactId>
    <version>4.4.0</version>
</dependency>
```

**spark-nlp-gpu:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-gpu_2.12</artifactId>
    <version>4.4.0</version>
</dependency>
```

**spark-nlp-silicon:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-silicon_2.12</artifactId>
    <version>4.4.0</version>
</dependency>
```

**spark-nlp-aarch64:**

```xml
<dependency>
    <groupId>com.johnsnowlabs.nlp</groupId>
    <artifactId>spark-nlp-aarch64_2.12</artifactId>
    <version>4.4.0</version>
</dependency>
```

**FAT JARs**

* CPU on Apache Spark 3.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-assembly-4.4.0.jar

* GPU on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-gpu-assembly-4.4.0.jar

* M1 on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-silicon-assembly-4.4.0.jar

* AArch64 on Apache Spark 3.0.x/3.1.x/3.2.x/3.3.x: https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-aarch64-assembly-4.4.0.jar

## Contributors
@Cabir40 @bunyamin-polat @danilojsl @dcecchini @Meryem1425 @C-K-Loan @agsfer @maziyarpanahi @jfernandrezj @jsl-builder @DevinTDHa @josejuanmartinez @aymanechilah

**Full Changelog**: https://github.com/JohnSnowLabs/spark-nlp/compare/4.3.2...4.4.0
